judge/views/register.py

```diff
@@ -43,12 +40,9 @@ def clean_email(self):
         if User.objects.filter(email=self.cleaned_data['email']).exists():
             raise forms.ValidationError(gettext('The email address "%s" is already taken. Only one registration '
                                                 'is allowed per address.') % self.cleaned_data['email'])
-        if '@' in self.cleaned_data['email']:
-            domain = self.cleaned_data['email'].split('@')[-1].lower()
-            if (domain in settings.BAD_MAIL_PROVIDERS or
-                    any(regex.match(domain) for regex in bad_mail_regex)):
-                raise forms.ValidationError(gettext('Your email provider is not allowed due to history of abuse. '
-                                                    'Please use a reputable email provider.'))
+        if is_email_address_bad(self.cleaned_data['email']):
+            raise forms.ValidationError(gettext('Your email provider is not allowed due to history of abuse. '
+                                                'Please use a reputable email provider.'))
```

Comment:
Let's refactor this common code. Potentially make a function called `validate_not_bad_email_domain` or something.

Reply:
This is what I had originally, but I dislike having a function `validate_...` that doesn't return anything and can only raise. I'm open to other suggestions though.

Reply:
On further thought, I'll change back to `validate_not_bad_email_domain`, since it looks like Django itself uses a similar construct.

Reply:
I think `throw_if_bad_email_domain` could work too, but don't feel like bikeshedding much more.